base/tests/unittests/com/google/idea/blaze/base/sync/workspace/ArtifactLocationDecoderTest.java

```diff
@@ -37,7 +37,7 @@ public class ArtifactLocationDecoderTest extends BlazeTestCase {
   public void testGeneratedArtifact() {
     ArtifactLocation artifactLocation =
         ArtifactLocation.builder()
-            .setRootExecutionPathFragment("/blaze-out/bin")
+            .setRootExecutionPathFragment("blaze-out/bin")
```

Comment:
AFAIU now using absolute path here will lead to `decoder.decode(artifactLocation)` being `"/blaze-out/bin/com/google/Bla.java`" instead of `EXECUTION_ROOT + "/blaze-out/bin/com/google/Bla.java"` (because of how Path.resolve works). We can artificially keep the old behavior (e.g. by removing all leading slashes to make path relative), but that sounds like a hack. 

Do you have any ideas/proposals on how this can be handled?

Reply:
Personally, I would prefer to support the old behavior, too (but prefer the new one) as we can't guarantee that other parts of the codebase always set correct relative path here. We probably should also log a warning in such case

Reply:
I have reverted this change by changing `ArtifactLocation.getExecutionRootRelativePath()`implementation. For me it still feels like a hack, though, PTAL.